src/telex.rs

```diff
@@ -38,6 +38,7 @@ where
             'r' => add_tone(&mut result, &ToneMark::HookAbove),
             'x' => add_tone(&mut result, &ToneMark::Tilde),
             'j' => add_tone(&mut result, &ToneMark::Underdot),
+            'z' => remove_tone(&mut result),
 
             'a' | 'e' | 'o' if contains_clean_char(&result, *ch) => {
```

Comment:
Since we've already performed a lowercase conversion on line 35, I think it'd be better to reuse that conversion instead of performing it again in the `contains_clean_char` function. So something like this:

```rs
let ch_lowercase = ch.to_ascii_lowercase();
let action_performed = match ch_lowercase {
   ....
   'a' | 'e' | 'o' if contains_clean_char(&result, ch_lowercase) => {
       ...
   }
}
```